code/game/mecha/mech_fabricator.dm

```diff
@@ -8,34 +15,21 @@
 	use_power = IDLE_POWER_USE
 	idle_power_usage = 20
 	active_power_usage = 5000
-	var/time_coeff = 1
-	var/component_coeff = 1
+	var/time_coeff = 1  // Determines build time for parts
+	var/component_coeff = 1  // Determines cost of parts
 	var/datum/research/files
 	var/fabricator_type = MECHFAB
-	var/id
-	var/sync = 0
-	var/part_set
-	var/datum/design/being_built
-	var/list/queue = list()
-	var/processing_queue = 0
-	var/screen = "main"
-	var/temp
-	var/list/part_sets = list(
-								"Cyborg",
-								"Cyborg Repair",
-								"Ripley",
-								"Firefighter",
-								"Odysseus",
-								"Gygax",
-								"Durand",
-								"H.O.N.K",
-								"Reticence",
-								"Phazon",
-								"Exosuit Equipment",
-								"Cyborg Upgrade Modules",
-								"Medical",
-								"Misc"
-								)
+	var/temp_search
+	var/wait_message // Used for popping up a wait message when the user presses sync
+	var/wait_message_timer
+	var/list/datum/design/matching_designs  // Used for designs that will appear after a search
+	var/datum/design/being_built // The design currently being built
+	var/list/queue = list()  // list of designs currently in the queue

```

Comment:
That's not what I mean.
(If my understanding of the DM object model is right) creating a list here in the object definition means that every single fabricator has this particular `list()` object as their `queue` field.
I.e. if you create 5 fabricators, and add something to a queue in one of them, then ALL of them will have that item added. Since the queue is literally shared.
Moving the initialization to something like `New()` would force DM to create a new `list()` for the queue each time a new fabricator is spawned in the world.

Does that make sense?

Reply:
Oh I get what you mean, that is not the case however. Each fabricator's `queue` will be unique to that specific machine. If you add something to one, it won't be added to all of them.